src/connector/src/sink/mod.rs

```diff
@@ -58,6 +60,36 @@ pub const SINK_TYPE_DEBEZIUM: &str = "debezium";
 pub const SINK_TYPE_UPSERT: &str = "upsert";
 pub const SINK_USER_FORCE_APPEND_ONLY_OPTION: &str = "force_append_only";
 
+#[derive(Debug, Clone)]
+pub struct SinkContext {
+    pub metrics: SinkMetrics,
+    pub info: SinkInfo,
+}
+pub type SinkContextRef = Arc<SinkContext>;
```

Comment:
Since `SinkContext` is owned by each sink executor, I think it's unnecessary to wrap it with an `Arc`. Besides, it is unnecessary to introduce an extra level of struct `SinkInfo`. A plain `SinkContext` following should be enough.

```
struct SinkContext {
    pub metrics: SinkMetrics,
    pub sink_param: SinkParam,
    pub actor_id: u32,
    pub fragment_id: u32,
}
```

We currently have a `SinkWriterParam`. We may consider putting fields added in this PR to it.

Reply:
Oh, I intend to pass the `SinkContextRef` to the inner sink connectors for the metrics and other related info. I guess `SinkWriterParam` will not do that.

Reply:
> Since SinkContext is owned by each sink executor, I think it's unnecessary to wrap it with an Arc. 

I use `Arc` for potential moves between threads, such as kafka background polling threads. The info here will not be changed and IMO there is little overhead.

Reply:
`SinkWriterParam` will be passed to the inner sink connector as a whole. 

In this PR, they are created at the same time when we create a new sink executor. And if we want to pass the `SinkContext` to kafka writer in a future PR, we will have to include it in `SinkWriterParam`. If they are created and consumed together, we can merge them together.

Correct me if I'm wrong. In future PR that monitors kafka, we are going to report kafka metrics to `SinkMetrics` with actor_id, fragment_id and sink_id as labels. If so, we don't have to move the whole `SinkContext` to the kafka polling thread. We shall create a local metrics with fixed labels and only pass the local metrics to the polling thread instead of passing the whole context.

Reply:
yes you are right, it is more appropriate to put the metrics in `SinkWriterParam `. will fix